plugins/git-plugin/src/main/groovy/org/rundeck/plugin/scm/git/ssh/SshjSession.groovy

```diff
@@ -0,0 +1,157 @@
+package org.rundeck.plugin.scm.git.ssh
+
+import groovy.transform.CompileStatic
+import net.schmizz.keepalive.KeepAliveProvider
+import net.schmizz.sshj.DefaultConfig
+import net.schmizz.sshj.SSHClient
+import net.schmizz.sshj.connection.channel.direct.Session
+import net.schmizz.sshj.transport.verification.PromiscuousVerifier
+import net.schmizz.sshj.userauth.keyprovider.KeyProvider
+import org.eclipse.jgit.errors.TransportException
+import org.eclipse.jgit.transport.OpenSshConfig
+import org.eclipse.jgit.transport.RemoteSession
+import org.eclipse.jgit.transport.URIish
+import org.eclipse.jgit.util.io.IsolatedOutputStream
+
+@CompileStatic
+class SshjSession implements RemoteSession {
+
+    Session session
+    SSHClient sshClient
+    URIish uri
+    private String privateKeyFile
+    Map<String, String> sshConfig
+    private OpenSshConfig config
+
+    SshjSession(URIish uri, Map<String, String> sshConfig, OpenSshConfig config, String privateKeyFile) {
+        this.sshConfig = sshConfig
+        this.config = config
+        this.privateKeyFile = privateKeyFile
+        this.uri = uri
+
+        this.sshClient = createConnection()
+
+    }
+
+    @Override
+    Process exec(String commandName, int timeout) throws IOException {
+        this.session = sshClient.startSession()
+        return new SshjProcess(commandName, timeout)
```

Comment:
To utilize the autoclosable feature of Session and Command:
```
try(Session session = sshClient.startSession()
    Session.Command command = session.exec(commandName)
) {
   return new SshProcess(command, timeout)
}
```

Reply:
I tried this but I got an error: Stream is closed, so maybe we can leave it as it is now